src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -213,13 +205,8 @@ private Task WriteMultiEndpointMessageAsync(ServiceMessage serviceMessage, Func<
             var routed = GetRoutedEndpoints(serviceMessage)?
                 .Select(endpoint =>
                 {
-                    if (_connectionContainers.TryGetValue(endpoint, out var connection))
-                    {
-                        return (e: endpoint, c: connection);
-                    }
-
-                    Log.EndpointNotExists(_logger, endpoint.ToString());
-                    return (e: endpoint, c: null);
+                    var connection = _routerEndpoints.endpoints.FirstOrDefault(e => e.Endpoint == endpoint.Endpoint).ConnectionContainer;
```

Comment:
as `GetRoutedEndpoints` is returned by customer, check if connection is `null` first?